packages/standard-components/src/Search.svelte

```diff
@@ -0,0 +1,158 @@
+<script>
+  import { getContext } from "svelte"
+  import { isEmpty } from "lodash/fp"
+  import {
+    Button,
+    DatePicker,
+    Label,
+    Select,
+    Toggle,
+    Input,
+  } from "@budibase/bbui"
+
+  const { API, styleable, DataProvider, builderStore } = getContext("sdk")
+  const component = getContext("component")
+
+  export let table = []
+  export let columns = []
+  export let pageSize = 50
+  export let noRowsMessage = "No Rows"
+
+  let rows = []
+  let loaded = false
+  let search = {}
+  let tableDefinition
+  let schema
+
+  // pagination
+  let page = 0
+
+  $: fetchData(table, page)
+  // omit empty strings
+  $: parsedSearch = Object.keys(search).reduce(
+    (acc, next) =>
+      search[next] === "" ? acc : { ...acc, [next]: search[next] },
+    {}
+  )
+
+  async function fetchData(table, page) {
+    if (!isEmpty(table)) {
+      const tableDef = await API.fetchTableDefinition(table)
+      schema = tableDef.schema
+      rows = await API.searchTable({
+        tableId: table,
+        search: parsedSearch,
+        pagination: {
+          pageSize,
+          page,
+        },
+      })
+    }
+    loaded = true
+  }
+
+  function nextPage() {
+    page += 1
+  }
+
+  function previousPage() {
+    page -= 1
+  }
+</script>
+
+<div use:styleable={$component.styles}>
+  <div class="query-builder">
+    {#if schema}
+      {#each columns as field}
+        <div class="form-field">
+          <Label extraSmall grey>{schema[field].name}</Label>
+          {#if schema[field].type === 'options'}
+            <Select secondary bind:value={search[field]}>
+              <option value="">Choose an option</option>
+              {#each schema[field].constraints.inclusion as opt}
+                <option>{opt}</option>
+              {/each}
+            </Select>
+          {:else if schema[field].type === 'datetime'}
+            <DatePicker bind:value={search[field]} />
+          {:else if schema[field].type === 'boolean'}
+            <Toggle text={schema[field].name} bind:checked={search[field]} />
+          {:else if schema[field].type === 'number'}
+            <Input type="number" bind:value={search[field]} />
+          {:else if schema[field].type === 'string'}
+            <Input bind:value={search[field]} />
+          {/if}
+        </div>
+      {/each}
+    {/if}
+    <div class="actions">
+      <Button
+        secondary
+        on:click={() => {
+          search = {}
+          page = 0
+        }}>
+        Reset
+      </Button>
+      <Button primary on:click={() => fetchData(table)}>Search</Button>
```

Comment:
Should we use a custom handler here that first resets the page to 0, and then calls `fetchData`? It would probably be good to reset `page` on each search. I think the `page` param is also missing here, but that wouldn't matter if it's being replaced with a different handler.

Reply:
It should be resetting the page - good catch 👍